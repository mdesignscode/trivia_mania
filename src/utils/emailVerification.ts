import { env } from '$env/dynamic/private';
import { EmailVerification } from '../models';
import nodemailer from 'nodemailer';

const sender = env.EMAIL_ADDRESS;
const pass = env.EMAIL_CODE;

// Configure the transporter
const transporter = nodemailer.createTransport({
        host: 'smtp.gmail.com', // e.g. smtp.gmail.com
        port: 587,
        secure: false, // true for 465, false for other ports
        auth: {
                user: sender,
                pass,
        },
});

// Send email function
export const sendEmail = async ({ email, username }: { username: string, email: string }) => {
        // generate email verification code
        const code = await generateEmailVerificationCode(email);

        // email body
        const textBody = `<p>Hi ${username},</p>
    <p>Thank you for signing up with <strong>Trivia Mania</strong>! To complete your registration, please use the verification code below:</p>
    <div class="code-box">${code}</div>
    <p>This code will expire in <strong>10 minutes</strong>. If you did not request this, you can safely ignore this email.</p>
    <p>Need help? <a href="mailto:${sender}">Contact our support team</a>.</p>
    <div class="footer">
      &copy; 2025 Trivia Mania. All rights reserved.
      <br>
      <a href="https://triviamania.vercel.app">https://triviamania.vercel.app</a>
    </div>`

        try {
                // send email
                await transporter.sendMail({
                        from: '"Trivia Mania" ejue1vs7n@mozmail.com', // sender address
                        to: email, // list of receivers
                        subject: 'Verify Your Email Address', // Subject line
                        html: textBody, // HTML body
                });

                return { status: 'success', data: code };
        } catch (error) {
                console.error('Error sending email:', error);
                return { status: 'fail', data: error.message };
        }
};


export async function generateEmailVerificationCode(email: string): Promise<string> {
        const code = Math.floor(100000 + Math.random() * 900000).toString(); // 6-digit code
        const expiresAt = new Date(Date.now() + 10 * 60 * 1000); // 10 minutes from now

        // existing verification should be updated
        const existingVerification = await EmailVerification.findOne({
                where: { email },
        });

        if (existingVerification) {
                existingVerification.setDataValue('code', code);
                existingVerification.setDataValue('expiresAt', expiresAt);
                await existingVerification.save();
                return code;
        }

        await EmailVerification.create({
                email,
                code,
                expiresAt,
                used: false,
        });

        return code;
}

export async function validateEmailVerificationCode(email: string, inputCode: string): Promise<{ success: boolean; message: string }> {
        const record = await EmailVerification.findOne({
                where: {
                        email,
                },
        });

        const failedResponse = { success: false, message: 'Invalid or expired verification code.' };

        if (record) {
                // if code hasn't expired yet
                const codeExpired = record.getDataValue('expiresAt') < new Date();
                if (codeExpired) return failedResponse;

                // is code correct
                const code = record.getDataValue('code');
                if (code !== inputCode) return failedResponse;

                // verified
                record.setDataValue('used', true)
                await record.save();

                return { success: true, message: 'Verification successful.' };
        }

        // record not found
        return failedResponse;
}

