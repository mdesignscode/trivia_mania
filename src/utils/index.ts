import { browser } from '$app/environment';

export function decodeHTMLEntities(text: string) {
        if (browser) {
                const element = document.createElement("div");
                element.innerHTML = text;
                return element.textContent || "";
        }
}

export const colorMap: { [key: string]: string } = {
        easy: "green",
        medium: "gold",
        hard: "red",
};

export function capitalize(str: string): string {
	if (!str) return '';
	return str.charAt(0).toUpperCase() + str.slice(1).toLowerCase();
}

