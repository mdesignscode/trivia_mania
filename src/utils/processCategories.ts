export type TQuestionData = {
        category: string;
        difficulty: string;
        count: number;
};
export type TAllCategories = Record<string, Record<string, TQuestionData>>;

export type TProcessResult = {
        allCategories: TAllCategories;
        allDifficulties: string[];
};

export function processCategories(
        categories: TQuestionData[],
        difficulties: string[]
): TProcessResult {
        const allCategories: TAllCategories = {};

        // map difficulties to categories
        categories.forEach(({ difficulty, count, category }) => {
                if (!allCategories[category]) allCategories[category] = {};
                allCategories[category][difficulty] = { difficulty, count, category };
        });

        // add "all difficulties" count for each category
        Object.keys(allCategories).forEach((key) => {
                const easy = allCategories[key].easy?.count || 0;
                const medium = allCategories[key].medium?.count || 0;
                const hard = allCategories[key].hard?.count || 0;
                const allCount = easy + medium + hard;

                allCategories[key]['all difficulties'] = {
                        difficulty: 'all difficulties',
                        count: allCount,
                        category: key
                };
        });

        // total "All Categories" aggregate
        const totalCategories: Record<string, TQuestionData> = {
                easy: { difficulty: 'easy', count: 0, category: 'All Categories' },
                medium: { difficulty: 'medium', count: 0, category: 'All Categories' },
                hard: { difficulty: 'hard', count: 0, category: 'All Categories' },
                'all difficulties': {
                        difficulty: 'all difficulties',
                        count: 0,
                        category: 'All Categories'
                }
        };

        Object.values(allCategories).forEach((catGroup) => {
                ['easy', 'medium', 'hard'].forEach((level) => {
                        if (catGroup[level]) {
                                totalCategories[level].count += catGroup[level].count;
                                totalCategories['all difficulties'].count += catGroup[level].count;
                        }
                });
        });

        allCategories['All Categories'] = totalCategories;

        return {
                allCategories,
                allDifficulties: [...difficulties]
        };
}

