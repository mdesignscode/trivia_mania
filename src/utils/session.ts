import { db } from '$lib/db';
import { encodeBase32LowerCaseNoPadding, encodeHexLowerCase } from '@oslojs/encoding';
import { sha256 } from '@oslojs/crypto/sha2';

export function generateSessionToken(): string {
        const bytes = new Uint8Array(20);
        crypto.getRandomValues(bytes);
        return encodeBase32LowerCaseNoPadding(bytes);
}

export function createSession(token: string, userId: number) {
        const sessionId = encodeHexLowerCase(sha256(new TextEncoder().encode(token)));
        const expiresAt = new Date(Date.now() + 1000 * 60 * 60 * 24 * 30); // 30 days

        db.prepare(
                'INSERT INTO session (id, user_id, expires_at) VALUES (?, ?, ?)'
        ).run(sessionId, userId, Math.floor(expiresAt.getTime() / 1000));
}

export function validateSessionToken(token: string) {
        const sessionId = encodeHexLowerCase(sha256(new TextEncoder().encode(token)));

        const row = db.prepare(`
    SELECT session.id, session.user_id, session.expires_at, user.username
    FROM session
    INNER JOIN user ON user.id = session.user_id
    WHERE session.id = ?
  `).get(sessionId);

        if (!row) {
                return { session: null, user: null };
        }

        const session = {
                id: row.id as string,
                userId: row.user_id as number,
                expiresAt: new Date(row.expires_at * 1000)
        };
        const user = {
                username: row.username as string
        };

        if (Date.now() >= session.expiresAt.getTime()) {
                db.prepare('DELETE FROM session WHERE id = ?').run(session.id);
                return { session: null, user: null };
        }

        return { session, user };
}

export function invalidateSession(sessionId: string) {
        db.prepare('DELETE FROM session WHERE id = ?').run(sessionId);
}

