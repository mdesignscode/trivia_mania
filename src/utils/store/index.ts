import globalStore from './global.svelte';
import { questionStore, playStore } from './playStore.svelte';

export {
        globalStore, playStore, questionStore,
}

