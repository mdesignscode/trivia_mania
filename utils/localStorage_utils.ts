export const QUESTIONS_POOL = "questionsPool",
  POOL_INDEX = "poolIndex",
  CURRENT_INDEX = "currentIndex",
  QUESTIONS_LIST = "questionsList",
  QUESTION_ANSWERED = "questionAnswered",
  ANSWERED_QUESTIONS = "answeredQuestions",
  PROGRESS = "progress",
  LAST_ANSWER = "lastAnswer",
  LAST_ANSWER_INDEX = "lastAnswerIndex",
  NEW_PARAMS = "newParams",
  CATEGORIES = "categories",
  DIFFICULTY = "difficulty",
  UNSAVED_DATA = "unsavedData",
  USERNAME = "username"

export function clearQuestionData() {
  localStorage.removeItem(UNSAVED_DATA);
  localStorage.removeItem(QUESTIONS_LIST);
  localStorage.removeItem(QUESTIONS_POOL);
  localStorage.removeItem(POOL_INDEX);
  localStorage.removeItem(CURRENT_INDEX);
  localStorage.removeItem(ANSWERED_QUESTIONS);
  localStorage.removeItem(LAST_ANSWER);
  localStorage.removeItem(LAST_ANSWER_INDEX);
  localStorage.removeItem(ANSWERED_QUESTIONS);
  localStorage.removeItem(NEW_PARAMS);
}
