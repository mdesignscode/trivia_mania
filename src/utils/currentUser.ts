import { redirect, type Cookies } from "@sveltejs/kit";
import { validateSessionToken } from "./session";

// check if user is already logged in
export const isSignedIn = async (token: string | undefined, route: string) => {
        // don't redirect auth pages
        const isAuthRoute = ['/login', '/signup'].includes(route);
        if (!token && isAuthRoute) return;

        // user not logged in
        if (!token) throw redirect(302, '/login');

        const { user } = await validateSessionToken(token);
        // user signed in
        if (user) throw redirect(302, '/');

        // no user
        if (!user && isAuthRoute) return;
        else throw redirect(302, '/login');
}

export const getUser = async (cookies: Cookies) => {
        const token = cookies.get('session');
        if (!token) {
                throw redirect(302, '/login');
        }

        const { session, user } = await validateSessionToken(token);
        if (!session || !user) {
                throw redirect(302, '/login');
        }

        const userData = user.dataValues;
        delete userData.password;

        return {
                user: userData,
        };
}

