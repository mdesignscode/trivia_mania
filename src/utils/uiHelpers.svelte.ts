import { globalStore } from 'store';
import type { TAllCategories } from './processCategories';

export function updateLocalPreferences(allCategories: TAllCategories, localStorage: Storage) {
        const localDifficulty = localStorage.getItem('difficulty') || '';
        const localSelectedCategories = localStorage.getItem('selectedCategories') || '';
        const selectedCategories = localSelectedCategories ? localSelectedCategories.split(',') : [];
        const availableCategories: string[] = [];
        let filteredCategories;

        // check availability of category
        selectedCategories.forEach((c) => {
                if (allCategories[c] && allCategories[c][localDifficulty]) {
                        availableCategories.push(c);
                }
        });

        // update local storage with available categories
        if (selectedCategories.some((c) => !availableCategories.includes(c)))
                localStorage.setItem('selectedCategories', availableCategories.join(','));

        // sync filters
        globalStore.difficulty = localDifficulty;
        globalStore.categories = availableCategories;

        if (localDifficulty) {
                filteredCategories = Object.values(allCategories)
                        .map((value) => value[localDifficulty])
                        .filter(Boolean);
        }
        return filteredCategories;
}

export function handleSelectCategory(category: string) {
        let selectedCategories = globalStore.categories.includes(category)
                ? globalStore.categories.filter((cat) => cat !== category)
                : [...globalStore.categories, category];
        if (category === 'All Categories') {
                if (!globalStore.categories.includes(category)) selectedCategories = [category];
        } else {
                if (globalStore.categories.includes('All Categories'))
                        selectedCategories = selectedCategories.filter((cat) => cat !== 'All Categories');
        }
        globalStore.categories = selectedCategories;
};

export function handleSetDifficulty(localStorage: Storage, allCategories: TAllCategories, difficulty?: string) {
        if (difficulty && typeof difficulty === 'string') {
                globalStore.difficulty = difficulty;
                localStorage.setItem('difficulty', difficulty);
        }
        const filteredCategories = Object.values(allCategories)
                .map((value) => value[globalStore.difficulty])
                .filter(Boolean);
        return filteredCategories;
};

export const handlePlay = (localStorage: Storage, goto: (path: string) => {}) => {
        const playUri = encodeURI(
                `/play?difficulty=${globalStore.difficulty}&categories=${globalStore.categories.join(',')}`
        );
        localStorage.setItem('difficulty', globalStore.difficulty);
        localStorage.setItem('selectedCategories', globalStore.categories.join(','));
        goto(playUri);
};

