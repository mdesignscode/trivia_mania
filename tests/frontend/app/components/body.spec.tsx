import Body from "@/components/body";
import { renderGlobalContext } from "@/utils/test_global_context";
import { screen } from "@/utils/test_utils";

describe("Body component", () => {
  it("Renders an introduction animation", async () => {
    const { baseElement } = renderGlobalContext(<Body>Mock child</Body>, {
      pageReady: false,
    });

    const container = await screen.findByTestId("intro-animation-container");
    expect(container).toBeInTheDocument();

    const introText = ["Trivia Mania", "by", "Marlon Baatjes"];

    for (let i = 0; i < introText.length; i++) {
      const word = introText[i];

      for (let j = 0; j < word.length; j++) {
        const letter = word[j]
        if (letter === " ") continue;
        const allMatchers = await screen.findAllByTestId(`${letter}_${j}`);
        for (const matcher of allMatchers) {
          expect(matcher).toBeInTheDocument();
        }
      }
    }

    expect(baseElement).toMatchSnapshot();
  });

  it("Renders <Navigation /> and current page content", async () => {
    const { baseElement } = renderGlobalContext(<Body>Mock child</Body>);

    const container = await screen.findByTestId("desktop-nav-container"),
      text = await screen.findByText("Mock child");

    expect(container).toBeInTheDocument();
    expect(text).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });
});
