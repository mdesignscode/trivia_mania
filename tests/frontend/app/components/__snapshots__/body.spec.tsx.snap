// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body component Renders <Navigation /> and current page content 1`] = `
<body>
  <div>
    <div>
      <div
        class="hidden md:block mx-auto w-full px-2 md:px-6 lg:px-8 bg-secondary text-light z-10 sticky top-0"
      >
        <div
          class="relative flex h-16 items-center justify-between"
        >
          <div
            class="flex flex-1 items-center justify-center md:items-stretch md:justify-start"
          >
            <div
              class="flex flex-shrink-0 items-center"
            >
              <a
                class="hover:bg-gray-700 rounded-md px-3 py-1"
                href="/"
              >
                <h1
                  class="text-white text-3xl"
                >
                  Trivia Mania
                </h1>
              </a>
            </div>
          </div>
          <div
            class="absolute inset-y-0 right-0 flex items-center pr-2 md:static md:inset-auto md:ml-6 md:pr-0"
            data-testid="desktop-nav-container"
          >
            <div
              class="hidden md:ml-6 md:block"
            >
              <div
                class="flex space-x-4"
              >
                <a
                  class="hover:bg-accent-100 hover:text-secondary rounded-md px-3 py-2 text-sm font-medium flex gap-2 content-center items-center"
                  href="/"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    height="25"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  Home
                </a>
                <a
                  class="hover:bg-accent-100 hover:text-secondary rounded-md px-3 py-2 text-sm font-medium flex gap-2 content-center items-center"
                  href="/game?difficulty=&categories="
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    height="25"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14.25 6.087c0-.355.186-.676.401-.959.221-.29.349-.634.349-1.003 0-1.036-1.007-1.875-2.25-1.875s-2.25.84-2.25 1.875c0 .369.128.713.349 1.003.215.283.401.604.401.959v0a.64.64 0 01-.657.643 48.39 48.39 0 01-4.163-.3c.186 1.613.293 3.25.315 4.907a.656.656 0 01-.658.663v0c-.355 0-.676-.186-.959-.401a1.647 1.647 0 00-1.003-.349c-1.036 0-1.875 1.007-1.875 2.25s.84 2.25 1.875 2.25c.369 0 .713-.128 1.003-.349.283-.215.604-.401.959-.401v0c.31 0 .555.26.532.57a48.039 48.039 0 01-.642 5.056c1.518.19 3.058.309 4.616.354a.64.64 0 00.657-.643v0c0-.355-.186-.676-.401-.959a1.647 1.647 0 01-.349-1.003c0-1.035 1.008-1.875 2.25-1.875 1.243 0 2.25.84 2.25 1.875 0 .369-.128.713-.349 1.003-.215.283-.4.604-.4.959v0c0 .333.277.599.61.58a48.1 48.1 0 005.427-.63 48.05 48.05 0 00.582-4.717.532.532 0 00-.533-.57v0c-.355 0-.676.186-.959.401-.29.221-.634.349-1.003.349-1.035 0-1.875-1.007-1.875-2.25s.84-2.25 1.875-2.25c.37 0 .713.128 1.003.349.283.215.604.401.96.401v0a.656.656 0 00.658-.663 48.422 48.422 0 00-.37-5.36c-1.886.342-3.81.574-5.766.689a.578.578 0 01-.61-.58v0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  Play
                </a>
                <a
                  class="hover:bg-accent-100 hover:text-secondary rounded-md px-3 py-2 text-sm font-medium flex gap-2 content-center items-center"
                  href="/scoreboard"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    height="25"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  Score Board
                </a>
                <div
                  class="flex gap-3"
                >
                  <a
                    class="hover:bg-accent-100 hover:text-secondary rounded-md px-3 py-2 text-sm font-medium flex gap-2 content-center items-center"
                    href="/users/mockId"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      height="25"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      width="25"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3 13.125C3 12.504 3.504 12 4.125 12h2.25c.621 0 1.125.504 1.125 1.125v6.75C7.5 20.496 6.996 21 6.375 21h-2.25A1.125 1.125 0 013 19.875v-6.75zM9.75 8.625c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125v11.25c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V8.625zM16.5 4.125c0-.621.504-1.125 1.125-1.125h2.25C20.496 3 21 3.504 21 4.125v15.75c0 .621-.504 1.125-1.125 1.125h-2.25a1.125 1.125 0 01-1.125-1.125V4.125z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                    Your Stats
                  </a>
                  <div
                    class="hover:bg-accent-100 hover:text-secondary rounded-md px-3 py-2 text-sm font-medium flex gap-2 content-center items-center"
                  >
                    <h1>
                      Mock user button
                    </h1>
                    <svg
                      aria-hidden="true"
                      fill="none"
                      height="25"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      width="25"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.982 18.725A7.488 7.488 0 0012 15.75a7.488 7.488 0 00-5.982 2.975m11.963 0a9 9 0 10-11.963 0m11.963 0A8.966 8.966 0 0112 21a8.966 8.966 0 01-5.982-2.275M15 9.75a3 3 0 11-6 0 3 3 0 016 0z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                    <h1>
                      Sign in
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <nav
        class="md:hidden bg-secondary text-light z-10 sticky top-0"
        data-headlessui-state=""
        data-testid="mobile-nav-container"
      >
        <div
          class="mx-auto max-w-7xl px-2 md:px-6 lg:px-8"
        >
          <div
            class="relative flex h-16 items-center justify-between"
          >
            <div
              class="absolute inset-y-0 left-0 flex items-center md:hidden"
            >
              <button
                aria-expanded="false"
                class="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
                data-headlessui-state=""
                data-testid="disclosure-button"
                id="headlessui-disclosure-button-:r0:"
                type="button"
              >
                <span
                  class="absolute -inset-0.5"
                />
                <span
                  class="sr-only"
                >
                  Open main menu
                </span>
                <svg
                  aria-hidden="true"
                  class="block h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="flex flex-1 items-center justify-center md:items-stretch md:justify-start"
            >
              <div
                class="flex flex-shrink-0 items-center"
              >
                <a
                  class="hover:bg-gray-700 rounded-md px-3 py-1"
                  href="/"
                >
                  <h1
                    class="text-white text-3xl"
                  >
                    Trivia Mania
                  </h1>
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
      Mock child
    </div>
  </div>
</body>
`;

exports[`Body component Renders an introduction animation 1`] = `
<body>
  <div>
    <div>
      <div
        class="col gap-4 my-auto"
        data-testid="intro-animation-container"
      >
        <div
          class="w-full gap-3 text-4xl px-2 md:text-7xl h-full flex justify-center items-center flex-wrap"
        >
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="T_0"
            style="animation-delay: 0s;"
          >
            T
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="r_1"
            style="animation-delay: 0.25s;"
          >
            r
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="i_2"
            style="animation-delay: 0.5s;"
          >
            i
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="v_3"
            style="animation-delay: 0.75s;"
          >
            v
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="i_4"
            style="animation-delay: 1s;"
          >
            i
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="a_5"
            style="animation-delay: 1.25s;"
          >
            a
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid=" _6"
            style="animation-delay: 1.5s;"
          >
             
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="M_7"
            style="animation-delay: 1.75s;"
          >
            M
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="a_8"
            style="animation-delay: 2s;"
          >
            a
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="n_9"
            style="animation-delay: 2.25s;"
          >
            n
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="i_10"
            style="animation-delay: 2.5s;"
          >
            i
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="a_11"
            style="animation-delay: 2.75s;"
          >
            a
          </span>
        </div>
        <div
          class="w-full gap-3 text-4xl px-2 md:text-7xl h-full flex justify-center items-center flex-wrap"
        >
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="b_0"
            style="animation-delay: 3s;"
          >
            b
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="y_1"
            style="animation-delay: 3.25s;"
          >
            y
          </span>
        </div>
        <div
          class="w-full gap-3 text-4xl px-2 md:text-7xl h-full flex justify-center items-center flex-wrap"
        >
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="M_0"
            style="animation-delay: 3.5s;"
          >
            M
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="a_1"
            style="animation-delay: 3.75s;"
          >
            a
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="r_2"
            style="animation-delay: 4s;"
          >
            r
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="l_3"
            style="animation-delay: 4.25s;"
          >
            l
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="o_4"
            style="animation-delay: 4.5s;"
          >
            o
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="n_5"
            style="animation-delay: 4.75s;"
          >
            n
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid=" _6"
            style="animation-delay: 5s;"
          >
             
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="B_7"
            style="animation-delay: 5.25s;"
          >
            B
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="a_8"
            style="animation-delay: 5.5s;"
          >
            a
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="a_9"
            style="animation-delay: 5.75s;"
          >
            a
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="t_10"
            style="animation-delay: 6s;"
          >
            t
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="j_11"
            style="animation-delay: 6.25s;"
          >
            j
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="e_12"
            style="animation-delay: 6.5s;"
          >
            e
          </span>
          <span
            class="animated-letter animate__animated animate__bounceInDown"
            data-testid="s_13"
            style="animation-delay: 6.75s;"
          >
            s
          </span>
        </div>
        <div
          class="bg-accent absolute -z-10 rounded-full w-64 h-64"
        />
      </div>
    </div>
  </div>
</body>
`;
